maintenance/main.py

```diff
@@ -296,15 +306,80 @@ def write_genome_data(self, ensembl_id):
+                # Get existing data
+                namespace = {}
+                with open(genome_data_path) as f:
+                    exec(f.read(), namespace)
+                existing_assembly_source = namespace["data"].get(
+                    "assembly_source", "ensembl"
+                )
+                if existing_assembly_source != "ensembl":
+                    logger.info(
+                        f"Skipping {sps_id} ({ensembl_id}): non-Ensembl assembly source"
+                    )
+                    return ("manual", None)
+            except Exception as e:
+                logger.warning(
+                    f"Error checking assembly source for {sps_id}: {e}. "
+                    "Proceeding with update."
+                )
+
+        # Get new data from Ensembl
         data = self.ensembl_client.get_genome_data(ensembl_id)
+
+        # Preserve existing assembly source or default to "ensembl"
+        if genome_data_path.exists():
+            try:
+                namespace = {}
+                with open(genome_data_path) as f:
+                    exec(f.read(), namespace)
+                data["assembly_source"] = namespace["data"].get(
+                    "assembly_source", "ensembl"
+                )
+            except Exception:
+                data["assembly_source"] = "ensembl"
+        else:
+            data["assembly_source"] = "ensembl"
+
+        # Add Ensembl version number if assembly source is Ensembl
+        if data["assembly_source"] == "ensembl":
+            data["assembly_build_version"] = str(self.ensembl_client.get_release())
+        else:
+            data["assembly_build_version"] = None
+
+        # Check if existing genome data exists and compare chromosome names
+        if genome_data_path.exists():
+            try:
+                # Get existing chromosome names
+                namespace = {}
+                with open(genome_data_path) as f:
+                    exec(f.read(), namespace)
+                existing_chroms = set(namespace["data"]["chromosomes"].keys())
+                new_chroms = set(data["chromosomes"].keys())
+
+                if existing_chroms != new_chroms:
+                    logger.warning(
+                        f"Skipping {sps_id} ({ensembl_id}): chromosome names mismatch."
+                    )
+                    return ("chrom_mismatch", (existing_chroms, new_chroms))
+            except Exception as e:
+                logger.warning(
+                    f"Error comparing chromosome names for {sps_id}: {e}. \
+                        Proceeding with update."
+                )
+
+        logger.info(f"Writing genome data for {sps_id} {ensembl_id}")
         code = f"data = {data}"
 
         # Format the code with Black so we don't get noisy diffs
-        with self.write(path) as f:
+        with self.write(genome_data_path) as f:
             f.write(black_format(code))
-        return data
+        return ("updated", None)
```

Comment:
Unless I'm missing something, this function can only return `("manual", None)` or `("updated", None)`?